### PR TITLE
translatesfx: add resourcedetection processor

### PR DESCRIPTION
**Pull request by pmcollins**

### CODE REVIEW

File: cmd/translatesfx/translatesfx/otel.go [hunk 1]
```diff
@@ -33,28 +33,33 @@ func saInfoToOtelConfig(cfg saCfgInfo) otelCfg {
...
+	const rd = "resourcedetection"
 	out := otelCfg{
 		ConfigSources: map[string]interface{}{
 			// TODO check if should be nil
 			"include": nil,
 		},
 		Receivers: receivers,
+		Processors: map[string]interface{}{
+			rd: map[string]interface{}{
+				"detectors": []string{"system", "env"},
```
Comment (mstumpfx) [inline cmd/translatesfx/translatesfx/otel.go:45]: I wonder if we should add the cloud provider resource detectors for generating their respective unique IDs (`AWSUniqueId`, `gcp_id`, `azure_resource_id`)?

Comment (pmcollins) [reply cmd/translatesfx/translatesfx/otel.go:45]: Hmm not sure -- at this point I'm targeting/automating the stuff that [this document](https://github.com/signalfx/splunk-otel-collector/blob/main/docs/signalfx-smart-agent-migration.md) recommends.

Thoughts @rmfitzpatrick?

**Review by dmitryax — COMMENTED** [review]
Could you change the commit message so it's clear that the change is applied to the translator?

**Comment by pmcollins** [comment]
> Could you change the commit message so it's clear that the change is applied to the translator?

Done

File: cmd/translatesfx/translatesfx/otel.go (hunk 1 above)
Comment (rmfitzpatrick) [reply cmd/translatesfx/translatesfx/otel.go:45]: I think the end goal for the tool is not just 1:1 config content but also all-around agent experience.  Believe the agent detects the cloud provider w/ its hostid utility, which adding all the resource detector targets would effectively do as well.  My concern would be the chattiness and continued noise w/ failed detection attempts though.

Comment (dmitryax) [reply cmd/translatesfx/translatesfx/otel.go:45]: I think we should be aligned with the default config where we have all of them added by default https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/config/collector/agent_config.yaml#L106

Comment (pmcollins) [reply cmd/translatesfx/translatesfx/otel.go]: Thanks guys for the feedback. I've added "gce", "ecs", "ec2", and "azure" resource detectors.

**Comment by pmcollins** [comment]
Looks like one of the tests is flaky, not sure how because failures only happen in CI and there's no log info, but I'm going to see if I can track it down and fix it.

**Comment by pmcollins** [comment]
Any further comments/questions/concerns?

**Review by mstumpfx — APPROVED** [review]
lgtm

**Pull request merged**